refresh.template.py

```diff
@@ -683,7 +694,36 @@ def _get_apple_active_clang():
     # Unless xcode-select has been invoked (like for a beta) we'd expect, e.g., '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang' or '/Library/Developer/CommandLineTools/usr/bin/clang'.
 
 
-def _apple_platform_patch(compile_args: typing.List[str]):
+@functools.lru_cache(maxsize=None)
+def _get_apple_active_swiftc():
+    """Get path to xcode-select'd swiftc version."""
+    return subprocess.check_output(
+        ('xcrun', '--find', 'swiftc'),
+        stderr=subprocess.DEVNULL, # Suppress superfluous error messages like "Requested but did not find extension point with identifier..."
+        encoding=locale.getpreferredencoding()
+    ).rstrip()
+
+
+def _apple_swift_patch(compile_args: typing.List[str]):
+    """De-Bazel(rule_swift) the command into something sourcekit-lsp can parse."""
+
+    # Remove build_bazel_rules_swift/tools/worker/worker
+    compile_args.pop(0)
+
+    # Expand -debug-prefix-pwd-is-dot
+    match = next((i for i,v in enumerate(compile_args) if v == "-Xwrapped-swift=-debug-prefix-pwd-is-dot"), None)
+    if match:
+        compile_args[match] = "-debug-prefix-map"
+        compile_args.insert(match + 1, os.environ["BUILD_WORKSPACE_DIRECTORY"] + "=.")
+
+    # Remove other -Xwrapped-swift arguments like `-ephemeral-module-cache` `-global-index-store-import-path`
+    # We could override index-store by config sourcekit-lsp
```

Comment:
What's our plan for integrating their indexing during build?
(Eventually we'll need to document how to configure things in the README, but that can wait until the end.)

Reply:
We can analyze commands in `aquery` and pre build .swiftmodule by striping `-emit-object`  in swift but that looks like it's going to be the second super slow header extractor...

For some small project fully build the target and cache the results by bazel ecosystem seems the best way but for some super huge project like bilibili it is impossible to wait a fully build before writing code.  So I am doing another set of rules and a vscode plugin for integrating indexing with building.  I will sync that once there is any progress

Reply:
For clarity: I wasn't meaning to propose that we force a full swift build up front when it isn't needed, but we should help people get sourcekit-lsp configured correctly, via instructions in the readme.

I'm guessing part of that will be pointing source kit-lsp to pick up the indexing output generated during bazel builds? And configuring bazel builds to generate that output if they don't already? 

I'd love to learn more about what you're working on with the plugin!

Reply:
[Here](https://github.com/xinzhengzhang/bis/blob/main/refresh_compile_commands.bzl) is my implementation.
I have wrapped the `hedron_refresh_compile_commands` and collect all the module info which depended by the target into outputs.
Not as painful as rebuilding .d file for find headers, we can share module objects with compiling and indexing in this way.